From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim=20D=C3=BCsterhus?= <tim@tideways-gmbh.com>
Date: Fri, 11 Sep 2026 22:45:33 +0200
Subject: [PATCH 17/17] URL: Fix `lxb_url_serialize_fragment()` without a query
 (#410)

The `lxb_url_serialize_fragment()` function checked if the query contained data
before correctly serializing the fragment. Check for the fragment instead.
---
 source/lexbor/url/url.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/source/lexbor/url/url.c b/source/lexbor/url/url.c
index 98ee304..faf553b 100644
--- a/source/lexbor/url/url.c
+++ b/source/lexbor/url/url.c
@@ -4915,7 +4915,7 @@ lxb_status_t
 lxb_url_serialize_fragment(const lxb_url_t *url,
                            lexbor_serialize_cb_f cb, void *ctx)
 {
-    if (url->query.data != NULL) {
+    if (url->fragment.data != NULL) {
         return cb(url->fragment.data, url->fragment.length, ctx);
     }
 
